lib/site-builder.js

```diff
@@ -93,26 +112,32 @@ SiteBuilder.launchBuilder = function(info, branch, builderConfig, done) {
   logger.log('sender:', info.sender.login);
 
   finishBuild = function(err) {
-    if (err !== undefined) {
-      logger.error(err);
-      logger.error(builderOpts.repoName + ': build failed');
-    } else {
-      logger.log(builderOpts.repoName + ': build successful');
-    }
-
-    // Provides https://pages.18f.gov/REPO-NAME/build.log as an indicator of
-    // latest status.
-    var newLogPath = path.join(
-      builder.configHandler.buildDestination, 'build.log');
-    fs.rename(buildLog, newLogPath, function(err) {
-      if (err !== null) {
-        console.error('Error moving build log from', buildLog, 'to',
-          newLogPath);
+    return new Promise(function(resolve, reject) {
+      // Provides https://pages.18f.gov/REPO-NAME/build.log as an indicator of
+      // latest status.
+      var newLogPath = path.join(
+        builder.configHandler.buildDestination, 'build.log');
+
+      if (err) {
+        logger.error(err.message ? err.message : err);
+        logger.error(builderOpts.repoName + ': build failed');
+      } else {
+        logger.log(builderOpts.repoName + ': build successful');
       }
-      logger.close(function() {
-        if (done) {
-          done(err);
+
+      fs.rename(buildLog, newLogPath, function(renameErr) {
+        var errMsg;
+
+        if (renameErr) {
+          errMsg = 'Error moving build log: ' + renameErr;
+          console.error(errMsg);
+          if (!err) {
+            reject(errMsg);
+          }
         }
+        logger.close(function() {
+          return err ? reject(err) : resolve();
+        });
       });
```

Comment:
You might look into using a Promise library to make this callback->Promise conversion easier, e.g.

* http://bluebirdjs.com/docs/api/promise.promisify.html
* http://documentup.com/kriskowal/q/#adapting-node